argilla-server/src/argilla_server/api/handlers/v1/datasets/datasets.py

```diff
@@ -90,29 +90,57 @@ async def list_current_user_datasets(
 
 @router.get("/datasets/{dataset_id}/fields", response_model=Fields)
 async def list_dataset_fields(
-    *, db: AsyncSession = Depends(get_async_db), dataset_id: UUID, current_user: User = Security(auth.get_current_user)
+    *,
+    db: AsyncSession = Depends(get_async_db),
+    telemetry_client: TelemetryClient = Depends(get_telemetry_client),
+    dataset_id: UUID,
+    current_user: User = Security(auth.get_current_user),
 ):
     dataset = await Dataset.get_or_raise(db, dataset_id, options=[selectinload(Dataset.fields)])
 
     await authorize(current_user, DatasetPolicy.get(dataset))
 
+    for field in dataset.fields:
+        await telemetry_client.track_crud_dataset_setting(
+            action="read", dataset=dataset, setting_name="fields", setting=field
+        )
```

Comment:
Do we know how these telemetry requests are done? Are they synchronous? UDP?

We should check that we are not spending a lot of time executing these requests so we are not adding additional time to the API endpoint requests.

Reply:
@jfcalvo underlying code can be found here. https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/utils/_telemetry.py